spec/openapi.yaml

```diff
@@ -83,10 +93,60 @@ paths:
             application/json:
               schema:
                 $ref: "#/components/schemas/FileInfo"
+    put:
+      summary: upload file or create folder
+      parameters:
+        - in: header
+          name: Content-Length
+          description: Length of file to upload
+          schema:
+            type: integer
+            example: 101342
+        - in: header
+          name: Content-Type
+          description: Type of uploaded media
+          schema:
+            type: string
+            default: application/binary
```

Comment:
Also technical nitpick. Is there such a mime type? Aren't you confusing it with application/octet-stream?

Reply:
`default` means - when client did not set. Or we can require this header, but with different response: 400 - Bad Request.
* bad mime type - 415 Unsupported Media Type
* no required header - 400 Bad Request